### PR TITLE
Adds support for custom Security Group rules

### PR DESCRIPTION
**Pull request by bradj**
## what

* Module now handles creating a security group on the users behalf

## why

* More flexible interface that allows the user to either create their own security group or simply supply a list of CIDR's

### CODE REVIEW

File: main.tf [hunk 1]
```diff
@@ -66,6 +67,27 @@ resource "aws_eip" "sftp" {
...
+  vpc_id = local.is_vpc ? var.vpc_id : null
+
+  rules = [
+    {
+      type        = "ingress"
+      from_port   = 22
+      to_port     = 22
+      protocol    = "tcp"
+      cidr_blocks = var.allowed_cidrs
+    }
```
Comment (osterman) [inline main.tf:85]: Nono, this goes against the spirit of the module. The objective is to expsoe the `rules` variable so the caller has unfettered ability to add rules for ingress/egress.

File: variables.tf [hunk 2]
```diff
@@ -38,6 +38,43 @@ variable "address_allocation_ids" {
...
+
+variable "security_group_rules" {
+  type = list(any)
+  default = [
+    {
+      type        = "ingress"
+      from_port   = 22
+      to_port     = 22
+      protocol    = "tcp"
+      cidr_blocks = ["10.0.0.0/16"]
```
Comment (Benbentwo) [inline variables.tf:67]: should this be an empty array by default?

File: variables.tf [hunk 3]
```diff
@@ -38,6 +38,43 @@ variable "address_allocation_ids" {
...
+
+variable "security_group_use_name_prefix" {
+  type        = bool
+  default     = false
+  description = "Whether to create a default Security Group with unique name beginning with the normalized prefix."
+}
+
+variable "security_group_rules" {
+  type = list(any)
+  default = [
```
Comment (Benbentwo) [inline variables.tf:61]: or this be an empty array by default?

File: variables.tf (hunk 2 above)
Comment (bradj) [reply variables.tf]: @Benbentwo I copied the pattern from [this repo](https://github.com/cloudposse/terraform-aws-elasticache-redis/blob/master/variables.tf#L30-L47).

It also has the `cidr_block` array populated with `0.0.0.0/0` but I don't like having it open to the world so leaving `cidr_block` with `[]` will show an error explaining it needs to be set. Not the greatest UX unfortunately but I think security should take precedence over UX.

**Review by Benbentwo — APPROVED** [review]

**Pull request merged**